### PR TITLE
Update client-go

### PR DESCRIPTION
**Pull request by bertinatto**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR updates `k8s.io/client-go` to the intermediate `kubernetes-1.17.0-rc.1`
version. This surfaced another problem, though: the auto-generated code
had to be updated as well. To achieve that, I imported
`k8s.io/code-generator` first and then executed `hack/update-generated-code.sh`.

I needed this change in order to use the new `v1.CSINode` in the
external-provisioner, which depends on the snapshotter.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by msau42** [comment]
One idea I had in https://github.com/kubernetes-csi/external-snapshotter/issues/186 was to separate the API + generated clients into a separate go module. I think that will help a lot with fixing dependency issues on the snapshotter

**Comment by xing-yang** [comment]
Code is generated differently now.  @yuxiangqian PTAL.
/hold

File: pkg/apis/volumesnapshot/v1beta1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -2,23 +2,26 @@
...
+
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Code generated by controller-gen. DO NOT EDIT.
+// Code generated by deepcopy-gen. DO NOT EDIT.
```
Comment (xing-yang) [inline pkg/apis/volumesnapshot/v1beta1/zz_generated.deepcopy.go]: We need to use controller-gen, not deepcopy-gen.

**Comment by bertinatto** [comment]
As we discussed in today's meeting, closing this in favour of https://github.com/kubernetes-csi/external-snapshotter/pull/202#issuecomment-560465403.

/close

**Pull request closed without merging**